packages/aztec.js/src/encoder/outputCoder.js

```diff
@@ -11,39 +11,56 @@ const { keccak256, padLeft } = require('web3-utils');
 const outputCoder = {};
 
 /**
- * Decode a note
+ * Decode an inputNote
  *
- * @method decodeNote
+ * @method decodeInputNote
  * @param {note} note - AZTEC note
  * @returns {Object[]} note variables - extracted variables: noteType, owner,
  * noteHash, gamma, sigma, ephemeral
  * // TODO: check what happens on develop; only decoding
  */
-outputCoder.decodeNote = (note) => {
-    const length = parseInt(note.slice(0x00, 0x40), 16);
-    let expectedLength;
+outputCoder.decodeInputNote = (note) => {
+    // expectedLength is 0xc0
     const noteType = parseInt(note.slice(0x40, 0x80), 16);
     const owner = `0x${note.slice(0x98, 0xc0)}`;
     const noteHash = `0x${note.slice(0xc0, 0x100)}`;
-    const metadataLength = parseInt(note.slice(0x100, 0x140), 16);
 
+    const ephemeral = null;
+    const gamma = bn128.decompressHex(note.slice(0x140, 0x180));
+    const sigma = bn128.decompressHex(note.slice(0x180, 0x1c0));
+
+    return {
+        noteType,
+        owner,
+        noteHash,
+        gamma,
+        sigma,
+        ephemeral,
+    };
+};
+
+/**
+ * Decode an output note
+ *
+ * @method decodeOutputNote
+ * @param {note} note - AZTEC note
+ * @returns {Object[]} note variables - extracted variables: noteType, owner,
+ * noteHash, gamma, sigma, ephemeral
+ * // TODO: check what happens on develop; only decoding
+ */
+outputCoder.decodeOutputNote = (note) => {
+    const noteType = parseInt(note.slice(0x40, 0x80), 16);
+    const owner = `0x${note.slice(0x98, 0xc0)}`;
+    const noteHash = `0x${note.slice(0xc0, 0x100)}`;
+    const noteDataLength = parseInt(note.slice(0x100, 0x140), 16);
     let ephemeral = null;
 
-    if (metadataLength === 0x137) {
-        // output note with metadata set
-        expectedLength = (0x20 * 4 + 0x20 * 2 + metadataLength).toString(16);
-        ephemeral = secp256k1.decompressHex(note.slice(0x1c0, 0x202));
-    } else if (metadataLength === 0x61) {
-        // ouputNote with no metadata set
-        expectedLength = 0xe1;
+    if (noteDataLength === 0x61) {
+        // ouputNote with no custom metaData set, expectedLength is 0xe1
         ephemeral = secp256k1.decompressHex(note.slice(0x1c0, 0x202));
     } else {
-        // inputNote
-        expectedLength = 0xc0;
-    }
-
-    if (length !== expectedLength) {
-        throw new Error(`unexpected note length of ${length}`);
+        // output note with custom metaData set, expectedLength is (0x20 * 4 + 0x20 * 2 + noteDataLength).toString(16);
+        ephemeral = secp256k1.decompressHex(note.slice(0x1c0, 0x202));
```

Comment:
This if/else seems superfluous?

Reply:
The only change between input note and output notes is the extraction of the ephemeral key and/or any additional metadata? If so, seems like you can simplify the API by simply always checking whether the length of the note contains/doesn't contain metadata.

Reply:
Good point, have done